pkgs/applications/audio/spotifyd/default.nix

```diff
@@ -0,0 +1,38 @@
+{ stdenv, fetchFromGitHub, rustPlatform, pkgconfig, openssl
+, withALSA ? true, alsaLib ? null
+, withPulseAudio ? false, libpulseaudio ? null
+}:
+
+rustPlatform.buildRustPackage rec {
+  name = "spotifyd-${version}";
+  version = "0.2.4";
+
+  src = fetchFromGitHub {
+    owner = "Spotifyd";
+    repo = "spotifyd";
+    rev = "v${version}";
+    sha256 = "0nmi33wvagmasxb59jwhlgdqlh8gfmqgwib16k1wx885qhx9yg30";
+  };
+
+  cargoPatches = [ ./noavxcrypto.patch ];
+
+  cargoSha256 = "09vial0psmh39j20ikmfldidl505jnpfnqa8ybszrf2pvrhszsgq";
+
+  cargoBuildFlags = [
+    "--no-default-features"
+    "--features"
+    "${stdenv.lib.optionalString withALSA "alsa_backend,"}${stdenv.lib.optionalString withPulseAudio "pulseaudio_backend,"}"
+  ];
+
+  buildInputs = [ pkgconfig openssl ]
+    ++ stdenv.lib.optional withALSA alsaLib
+    ++ stdenv.lib.optional withPulseAudio libpulseaudio;
+
+  meta = with stdenv.lib; {
+    description = "An open source Spotify client running as a UNIX daemon.";
+    homepage = https://github.com/Spotifyd/spotifyd;
+    license = with licenses; [ gpl3 ];
+    maintainers = [ maintainers.jmgrosen ];
+    platforms = platforms.all;
```

Comment:
I think `buildRustPackage` sets this for us, also I don't think this would build for all platforms anyway.

Reply:
I can't find anywhere that suggests `buildRustPackage` does this -- at the least, the example in the nixpkgs manual still has a meta section. I added (what I think should be) darwin support in my latest commit, so I think `platforms.all` is appropriate too.

Reply:
> I can't find anywhere that suggests `buildRustPackage` does this -- at the least, the example in the nixpkgs manual still has a meta section.

Ahh, yes that is correct.

> . I added (what I think should be) darwin support in my latest commit, so I think `platforms.all` is appropriate too.

I think [`all`](https://github.com/NixOS/nixpkgs/blob/a44784e81181c971a41c588d93a6cf4bbd1a394c/lib/systems/doubles.nix#L8) is way too broad in this context. If you want darwin and linux support you can use `unix`.